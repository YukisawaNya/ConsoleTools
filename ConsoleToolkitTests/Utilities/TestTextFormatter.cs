using ApprovalTests;
using ApprovalTests.Reporters;
using ConsoleToolkit.Utilities;
using ConsoleToolkitTests.TestingUtilities;
using NUnit.Framework;

namespace ConsoleToolkitTests.Utilities
{
    [TestFixture]
    [UseReporter(typeof (CustomReporter))]
    public class TestTextFormatter
    {
        [Test]
        public void BlocksWithEqualLengthAreMergedProperly()
        {
            var left =
                @"First line
Second line";
            var right =
                @"A
B";
            Approvals.Verify(TextFormatter.MergeBlocks(left, 25, right));
        }

        [Test]
        public void BlocksWhereTheLeftSideIsLongerAreMergedProperly()
        {
            var left =
                @"First line
Second line";
            var right = @"A";
            Approvals.Verify(TextFormatter.MergeBlocks(left, 25, right));
        }

        [Test]
        public void BlocksWhereTheRightSideIsLongerAreMergedProperly()
        {
            var left =
                @"First line
Second line";
            var right =
                @"A
B
C
D
E";
            Approvals.Verify(TextFormatter.MergeBlocks(left, 25, right));
        }
    }
}
