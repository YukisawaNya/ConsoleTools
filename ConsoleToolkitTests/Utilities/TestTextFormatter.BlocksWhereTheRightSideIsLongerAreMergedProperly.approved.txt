First line               A
Second line              B
                         C
                         D
                         E
